src/Model/Post.php

```diff
@@ -328,7 +328,7 @@ public function is_private() {
 		/**
 		 * Published content is public, not private
 		 */
-		if ( 'publish' === $this->data->post_status ) {
+		if ( 'publish' === $this->data->post_status && $this->post_type_object && $this->post_type_object->publicly_queryable ) {
```

Comment:
@kellenmace this is interesting. 

I think if `public` is true, we need to respect it first. 

Because post types _can_ be `public => true, publicly_queryable => false`, such as the `page` post_type: https://github.com/WordPress/WordPress/blob/8447683474ebdee773fc4fca7b3e44dea45060f4/wp-includes/post.php#L45-L68, which appears to be breaking a lot of tests right now 🤔

Reply:
@kellenmace I pushed up a change so it's now:

```php
if ( 'publish' === $this->data->post_status && $this->post_type_object && ( true === $this->post_type_object->public || true === $this->post_type_object->publicly_queryable ) ) {
  return false;
}
```

This means that if _either_ `public` _or_ `publicly_queryable` are true, then WPGraphQL should treat the entries as public.

This would allow for "utility" post_types, such as a Slideshow post_type or an FAQ post_type, etc to be registered as `public=>false, publicly_queryable=>true` and the entries would not get a public url, _but_ would be queryable in WPGraphQL, so they could still be used to compose public facing pages, even without having their own public urls.